### ISSUE TITLE
DevelopmentML estimator

### ISSUE THREAD

**Issue by jbogaardt** [casact/chainladder-python#129]
With `sparse` triangle support,  `index` and `column` values, and tight integration with `sklearn` already available, it shouldn't take too much effort to blend ML regression techniques with the convenient reserving properties of the `chainladder` package.

Refer to the ideas presented here - 
https://www.casact.org/education/clrs/2016/presentations/AR-1_1.pdf

Something like:
``` python
cl.DevelopmentML(
   estimator=sklearn.ensemble.RandomForestRegressor()
).fit(cl.load_sample('prism')).ldf_
```

We'd need some way of providing control over feature selection and feature engineering, but otherwise seems doable.

**Comment by jbogaardt** [comment]
GLM is such an important framework in actuarial science it made sense to develop a separate `TweedieGLM` development esimtator just for that.  Under the hood it uses the `DevelopmentML` estimator.  The `TweedieGLM` is similar to the R ChainLadder `glmReserve`, but where we differ is:
1. I haven't fleshed out the prediction error, and they are thorough there
2. Instead I focused on flexibility in adding covariates (jurisdiction, cause of loss, limits, report lag, etc) to the model. In theory it could be used for policy/claim level loss reserving since the sparse backend allows for high throughput of data in chainladder.  Flexibility come by way of the `patsy` package, but its a reasonably small dependency for a lot of flexibility.

Pre-release docs:
https://chainladder-python.readthedocs.io/en/latest/modules/development.html#tweedieglm

**Comment by andrejakobsen** [comment]
Are there plans to continue developing this feature? I could try to contribute, if I find the time.

**Comment by jbogaardt** [comment]
Hello @andrejakobsen, welcome! Yes, I would like to see this developed out more. Time is definitely the most constraining factor. If you have ideas, I would like to get them on board.
